src/app/core/cms/halremote.cache.ts

```diff
@@ -47,22 +47,28 @@ export class HalRemoteCache {
     if (Env.CMS_TTL === 0) {
       return valObservable;
     }
-    this.OBSERVABLES[key] = valObservable.share();
-    this.OBSERVABLES[key].subscribe(
-      (val: any) => {
-        if (Env.CMS_USE_LOCALSTORAGE && window && window.localStorage) {
-          window.localStorage.setItem(key, JSON.stringify(val));
-          window.localStorage.setItem(`expires.${key}`, JSON.stringify(new Date().getTime()));
-        } else {
-          this.VALUES[key] = val;
-          this.EXPIRES[key] = new Date().getTime();
-        }
-        setTimeout(() => { delete this.OBSERVABLES[key]; }, 0);
-      },
-      (err: any) => {
-        setTimeout(() => { this.del(key); }, 0);
+
+    // monitor observable for value
+    let gotValue = false;
+    this.OBSERVABLES[key] = valObservable.share().map(val => {
```

Comment:
Spent a long time trying to refactor this method, to work with the auth-token changing.  The keys were (1) don't subscribe, just `map`, and (2) call `share()` before anything else.  Otherwise, I got strange results where it cached the result of the token-lookup-observable, but ran the HTTP request every single time.  (Even thinking about this enough to write this comment makes my head hurt).